src/processors/physics.rs

```diff
@@ -0,0 +1,160 @@
+//! Physics processor based on `nphysics3d`
+
+extern crate ncollide;
+extern crate nphysics3d;
+
+use std::ops::{Deref, DerefMut};
+use std::sync::{Arc, Mutex};
+
+use processors::transform::LocalTransform;
+
+use context::Context;
+use ecs::{Component, Join, Processor, RunArg, VecStorage};
+
+// Reëxport `ncollide` shapes
+pub use self::ncollide::shape::{Ball, Plane, Cuboid, Capsule, Cone, Cylinder, ConvexHull,
+                                MinkowskiSum, AnnotatedMinkowskiSum, AnnotatedPoint, Reflection,
+                                Compound, BaseMesh, BaseMeshElement, TriMesh, TriMesh3, Polyline,
+                                Segment, Triangle, Torus, CompositeShape, SupportMap};
+
+// Also reëxport some `nphysics3d` objects
+pub use self::nphysics3d::object::{RigidBody, RigidBodyHandle, Sensor, SensorHandle};
+use self::nphysics3d::world::World as PhysicsWorld;
+
+
+/// Wrapper around `RigidBodyHandle` that impls some convenience traits.
+/// NOTE: It also impls `Send`/`Sync`, which makes it unsafe to use in
+/// multiple `World`'s.
+/// TODO: Investigate moving to a thread-safe version.
+pub struct PhysicsComponent(pub RigidBodyHandle<f32>);
+
+unsafe impl Sync for PhysicsComponent { }
+unsafe impl Send for PhysicsComponent { }
+
+impl Component for PhysicsComponent {
+    type Storage = VecStorage<PhysicsComponent>;
+}
+
+impl Deref for PhysicsComponent {
+    type Target = RigidBodyHandle<f32>;
+    fn deref(&self) -> &RigidBodyHandle<f32> {
+        &self.0
+    }
+}
+
+impl DerefMut for PhysicsComponent {
+    fn deref_mut(&mut self) -> &mut RigidBodyHandle<f32> {
+        &mut self.0
+    }
+}
+
+
+/// Wrapper around `nphysics`'s `World` that impls some convenience traits.
+/// NOTE: It also impls `Send`/`Sync`, which makes it unsafe to use in
+/// multiple threads.
+/// TODO: Investigate moving to a thread-safe version.
+pub struct World(PhysicsWorld<f32>);
```

Comment:
I would change this `World` to be `PhysicsWorld` and just use `nphysics3d::world::World` inside the new-type.